common/sketch/src/main/java/org/apache/spark/util/sketch/DefaultBloomFilter.java

```diff
@@ -0,0 +1,220 @@
+        throw new RuntimeException("Only support utf-8 string", e);
+      }
+    } else {
+      long longValue;
+
+      if (item instanceof Long) {
+        longValue = (Long) item;
+      } else if (item instanceof Integer) {
+        longValue = ((Integer) item).longValue();
+      } else if (item instanceof Short) {
+        longValue = ((Short) item).longValue();
+      } else if (item instanceof Byte) {
+        longValue = ((Byte) item).longValue();
+      } else {
+        throw new IllegalArgumentException(
+          "Support for " + item.getClass().getName() + " not implemented"
+        );
+      }
+
+      int h1 = Murmur3_x86_32.hashLong(longValue, 0);
+      int h2 = Murmur3_x86_32.hashLong(longValue, h1);
+      return (((long) h1) << 32) | (h2 & 0xFFFFFFFFL);
+    }
+  }
+
+  private static long hashBytesToLong(byte[] bytes) {
+    int h1 = Murmur3_x86_32.hashUnsafeBytes(bytes, Platform.BYTE_ARRAY_OFFSET, bytes.length, 0);
+    int h2 = Murmur3_x86_32.hashUnsafeBytes(bytes, Platform.BYTE_ARRAY_OFFSET, bytes.length, h1);
+    return (((long) h1) << 32) | (h2 & 0xFFFFFFFFL);
+  }
+
+  @Override
+  public boolean put(Object item) {
+    long bitSize = bits.bitSize();
+    long hash64 = hashObjectToLong(item);
+    int h1 = (int) (hash64 >> 32);
+    int h2 = (int) hash64;
+
+    boolean bitsChanged = false;
+    for (int i = 1; i <= numHashFunctions; i++) {
+      int combinedHash = h1 + (i * h2);
+      // Flip all the bits if it's negative (guaranteed positive number)
+      if (combinedHash < 0) {
+        combinedHash = ~combinedHash;
+      }
+      bitsChanged |= bits.set(combinedHash % bitSize);
+    }
+    return bitsChanged;
+  }
+
+  @Override
+  public boolean mightContain(Object item) {
+    long bitSize = bits.bitSize();
+    long hash64 = hashObjectToLong(item);
+    int h1 = (int) (hash64 >> 32);
+    int h2 = (int) hash64;
+
+    for (int i = 1; i <= numHashFunctions; i++) {
+      int combinedHash = h1 + (i * h2);
+      // Flip all the bits if it's negative (guaranteed positive number)
+      if (combinedHash < 0) {
+        combinedHash = ~combinedHash;
+      }
+      if (!bits.get(combinedHash % bitSize)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public boolean isCompatible(BloomFilter other) {
+    if (other == null) {
+      return false;
+    }
+
+    if (!(other instanceof DefaultBloomFilter)) {
+      return false;
+    }
+
+    DefaultBloomFilter that = (DefaultBloomFilter) other;
+    return this.bitSize() == that.bitSize() && this.numHashFunctions == that.numHashFunctions;
+  }
+
+  @Override
+  public BloomFilter mergeInPlace(BloomFilter other) {
+    if (!isCompatible(other)) {
+      throw new IllegalArgumentException("Can't merge incompatible bloom filter");
+    }
+
+    DefaultBloomFilter that = (DefaultBloomFilter) other;
+    this.bits.putAll(that.bits);
+    return this;
+  }
+
+  static final class BitArray {
```

Comment:
I'd move this out of this file and just have a BitSet class. We should have a test suite for this too.